www/js/lib/uiUtil.js

```diff
@@ -51,15 +51,12 @@ define(rqDef, function(settingsStore) {
             // DEV: Note that webpMachine is single threaded and will reject an image if it is busy
             // However, the loadImagesJQuery() function in app.js is sequential (it waits for a callback
             // before processing another image) so we do not need to queue WebP images here
-            webpMachine.decode(content).then(function (uri) {
-                // DEV: WebpMachine.decode() returns a data: URI
-                // We callback before the node is set so that we don't incur slow DOM rewrites before processing more images
-                if (callback) callback(uri);
-                node.setAttribute(nodeAttribute, uri);
-            }).catch(function (err) {
-                console.error('There was an error decoding image in WebpMachine', err);
-                if (callback) callback();
-            });
+            var canvas = document.createElement("canvas");
+            webpMachine.decodeToCanvas(canvas, content);
+            // TODO copy all the img attributes to the new canvas, and remove the img node afterwards
+            node.parentNode.appendChild(canvas);
+            // TODO we don't handle images sequentially
+            if (callback) callback();
```

Comment:
@mossroy  In case it's useful, here's my code so far as a suggestion. Alternatives I've tried that I think don't work are commented out. This works in IceCat, but I've failed to be able to set the canvas size correctly (see commented out code for what I tried).
```suggestion
            var canvas = document.createElement("canvas");
            webpMachine.decodeToCanvas(canvas, content).then(function () {
                if (canvas.msToBlob) {
                    insertBlob(canvas.msToBlob());
                } else if (canvas.mozGetAsFile) {
                    
                    // ** DEV: This is mozGetAsFile method - but results in blank images **
                    // var f = canvas.mozGetAsFile('f.png');
                    // insertBlob(f);

                    // ** DEV: This is Insert canvas method - but if we set the height or width, the imgaes get blanked **
                    if (callback) callback(); // Calling back as soon as possible speeds up extraction
                    // canvas.setAttribute('width', node.getAttribute('width'));
                    // canvas.setAttribute('height', node.getAttribute('height'));
                    node.parentNode.replaceChild(canvas, node);

                } else {
                    canvas.toBlob(function (imgBlob) {
                        insertBlob(imgBlob);
                    });
                }
            }).catch(function (err) {
                console.error('Error converting WebP image!', err);
            });
```

Reply:
@Jaifroid i also had initial troubles setting the canvas dimensions – but i found success with the following technique https://github.com/chase-moskal/webp-hero/blob/f75a7593a1dd1271ef9b075d9c1611bc9c2466ea/source/webp-machine.ts#L153

you can see this in action in the live webp-hero demo: https://webp-hero.chasemoskal.com/?force?useCanvasElements

a friend of mine pointed out that canvas elements stop scrolling from working while the user hovers over a canvas -- i'm intending to add the css rule `pointer-events: none;` to mitigate this.

i can expose the webp-hero helper functions as statics on WebpMachine in the next version of webp-hero, i should be able to get to that in a few hours :+1:

Reply:
That's incredibly helpful @chase-moskal . I was actually thinking to check your source code as you obviously need to set dimensions too when polyfilling the whole document. We'll try the style trick!

Reply:
@chase-moskal That works! Thank you! Note that I had to add the 'px' explicitly or else the images were still the wrong size, as I think a simple number (even as a string) is not accepted by the `style.width/height` attributes. I'm also using the `node.parentNode.replaceChild(newNode, oldNode)` method to replace the old node with the canvas, because of the need to support IE11 in our code still.
```
canvas.style.width = node.getAttribute('width') + 'px';
canvas.style.height = node.getAttribute('height') + 'px';
node.parentNode.replaceChild(canvas, node);
```

![image](https://user-images.githubusercontent.com/4304337/153745778-d58d7263-712f-4faf-903a-375913d649dc.png)

Reply:
@mossroy I updated the suggestion above with latest working code.

Reply:
@Jaifroid @mossroy 

:tada: please see [this comment](https://github.com/kiwix/kiwix-js/issues/835#issuecomment-1038000222) in the issue

Reply:
Many thanks @chase-moskal 
@Jaifroid : feel free to commit your work-in-progress on this branch/PR and continue it (if you are willing to).

Reply:
OK, I've pushed my changes. I haven't yet switched to latest Web-Hero to use the new static function. If you want to have a go at that, please do, as I won't have much more time today.